Common/include/geometry/primal_grid/CPrimalGridFEM.hpp

```diff
@@ -213,8 +213,26 @@ class CPrimalGridFEM final: public CPrimalGrid {
    */
   void GetCornerPointsAllFaces(unsigned short &numFaces,
                                unsigned short nPointsPerFace[],
-                               unsigned long  faceConn[6][4]) override;
+                               unsigned long  faceConn[6][4]) const override;
+private:
+  /*!
+   * \brief Get the number of faces of the element.
+   * \param[in] elementType - element type
+   * \return number of faces
+   */
+  static inline unsigned short nFacesOfElementType(unsigned short elementType) {
+    switch (elementType) {
+      case TRIANGLE: return 3;
+      case QUADRILATERAL: return 4;
+      case TETRAHEDRON: return 4;
+      case PYRAMID: return 5;
+      case PRISM: return 5;
+      case HEXAHEDRON: return 6;
```

Comment:
Note1: in option_structure.hpp we have
```c++
const unsigned short N_POINTS_TRIANGLE = 3;
```
so we could do the same for N_FACES_*

Note2 off-topic: I already wanted to plug some enum class change here, but the VTK element types actually have some fixed numbers attached to them ... so one might even state that over there.

Reply:
In addition to note 1, the classes `CLine`, `CTriangle` etc have private static members `nFaces`, `nNodes` replicating the information another time. We could also make those public (and constexpr), and use them in `nFacesOfElementType` as well as to replace `N_POINTS_TRIANGLE` wherever possible. I'll try this if you don't object @TobiKattmann @WallyMaier @pcarruscag .

Reply:
The objective of those classes being virtual (well, any class) is that you don't need their declaration everywhere.
So you would move N_FACES_TRIANGLE to e.g. option_structure (or to primal grid) and include it in CTriangle.cpp, and not include CTriangle everywhere you need N_FACES_TRIANGLE.

That family of classes could use some work, it is taking the worst of two worlds in some cases.
We can talk about that if you're interested in changing some of the older parts of SU2 :smile:

Reply:
Ok, so I guess the only reasonable approach is then for e.g. for `TRIANGLE` to add `const unsigned short N_FACES_TRIANGLE=3` to option structure.hpp and then find - replace all `3`s in the codebase 🤝

Reply:
:facepalm:

Reply:
I added `N_FACES_*` and use the constants in option_structure.hpp in `nFacesOfElementType` and for the static members `nNodes`/`nFaces` of CLine etc. The numbers are hard-coded at many other locations, both in CLine etc (e.g. dimensions of `Faces`, `Neighbor_Nodes`,...) and outside. 
I am interested in how these classes could be improved  (I'll ask in some developer meeting, though maybe I cannot attend next week), but regarding this PR I'd like to finish it here ;)

Reply:
:+1: Regarding finishing PR's, I think it's fair to assume that if someone approves you don't have to go over their wishlist (unless they express the approval is conditional on that).

Ok, we can also talk via slack if you want. Or open a discussion, although it would take me longer to explain via text than to code the small changes I'm thinking :smile: